File: backend/app/schemas/resume.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ResumeRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    candidate_id: int
    file_name: str
    file_path: str
    file_type: str | None
    file_size: int | None
    raw_text: str | None
    parse_status: str
    parse_error: str | None
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/resume.ts
export interface Resume {
  id: number;
  candidate_id: number;
  file_name: string;
  file_path: string;
  file_type?: string;
  file_size?: number;
  raw_text?: string;
  parsed_json?: {
    name?: string;
    phone?: string;
    email?: string;
    current_title?: string;
    years_of_experience?: number;
    highest_education?: string;
    current_city?: string;
    skills?: string[];
  };
  parse_status: string;
  parse_error?: string;
  created_at: string;
  updated_at: string;
}
